core/src/main/java/org/apache/iceberg/encryption/WrappedEncryptionKey.java

```diff
@@ -43,4 +45,12 @@ public ByteBuffer wrappedKey() {
   public long timestamp() {
     return timestamp;
   }
+
+  public ByteBuffer key() {
+    return keyBytes;
+  }
+
+  public void setUnwrappedKey(ByteBuffer key) {
```

Comment:
This should not contain a key. It's fine to cache, but cache in `unwrapKey` instead.